SerialPrograms/Source/PokemonSV/Inference/ItemPrinter/PokemonSV_ItemPrinterPrizeReader.cpp

```diff
@@ -117,7 +124,76 @@ std::array<std::string, 10> ItemPrinterPrizeReader::read(
 }
 
 
+std::array<int16_t, 10> ItemPrinterPrizeReader::read_quantity(
+    Logger& logger, AsyncDispatcher& dispatcher,
+    const ImageViewRGB32& screen
+) const{
+    int8_t total_rows = 10;
+
+    // determine whether to use normal or bonus boxes for OCR
+    double total_average_sum_normal = 0;
+    double total_average_sum_bonus = 0;
+    for (int8_t i = 0; i < total_rows; i++){
+        total_average_sum_normal += average_sum_filtered(screen, m_boxes_normal[i]);
+        total_average_sum_bonus += average_sum_filtered(screen, m_boxes_bonus[i]);
+    }
 
+    // std::cout << "total_average_sum_normal: " << std::to_string(total_average_sum_normal) << std::endl;
+    // std::cout << "total_average_sum_bonus: " << std::to_string(total_average_sum_bonus) << std::endl;
+
+
+    std::array<int16_t, 10> results;
+    std::vector<std::unique_ptr<AsyncTask>> tasks(total_rows);
+    for (int8_t i = 0; i < total_rows; i++){
+        // ImageRGB32 filtered = to_blackwhite_rgb32_range(
+        //     extract_box_reference(screen, m_boxes_bonus_quantity[i]),
+        //     0xff808000, 0xffffffff,
+        //     true
+        // );
+        // filtered.save("DebugDumps/test"+ std::to_string(i) +".png");   
+
+        tasks[i] = dispatcher.dispatch([&, i]{
+            if (total_average_sum_normal > total_average_sum_bonus){

```

Comment:
Basic programming question: Is there a more elegant way of formatting the if statement here?

I was thinking of something like this. But without `boxes` copying the whole array.
```
std::array<ImageFloatBox, 10> boxes = m_boxes_bonus_quantity;
if (total_average_sum_normal > total_average_sum_bonus){
	boxes = m_boxes_normal_quantity;
}

for (int8_t i = 0; i < total_rows; i++){
       tasks[i] = dispatcher.dispatch([&, i]{
            results[i] = read_number(logger, screen, boxes[i]);            
        });
    }
```

Reply:
you can make boxes be a reference ?
std::array<ImageFloatBox, 10>**&** boxes = m_boxes_bonus_quantity;

Reply:
When I try making `boxes` a reference, I get this error: 

C2440: 'initializing': cannot convert from 'const std::array<PokemonAutomation::ImageFloatBox,10>' to 'std::array<PokemonAutomation::ImageFloatBox,10> &'

`const std::array<ImageFloatBox, 10>& boxes = m_boxes_bonus_quantity;` doesn't work either: 

error C2678: binary '=': no operator found which takes a left-hand operand of type 'const std::array<PokemonAutomation::ImageFloatBox,10>' (or there is no acceptable conversion)

Also, would a reference even work? Since references can't be reassigned? Once I assign `boxes = m_boxes_bonus_quantity;`, then I can't reassign it to `m_boxes_normal_quantity`, if the if-statement is true, correct?

Reply:
oh yes you're right, must have been tired this morning. You could do a method that return "const std::array<ImageFloatBox, 10>&"

Then inside it, based on "total_average_sum_normal" and "total_average_sum_bonus", you return one of the two array ? I guess it could be a lambda too. But honeslty, i feel current version is okay ?

Reply:
Thanks! Yeah, a separate function/lambda works. It looks like a ternary operator works as well.
If you feel it's fine as it is, then I'll just leave it.

Reply:
I would use the ?: ternary operator.